src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1666,35 +1546,10 @@ internal static Process Start(ProcessStartInfo startInfo, ProcessWindowStyle win
                     shellExecuteInfo.lpDirectory = Marshal.StringToHGlobalUni(startInfo.WorkingDirectory);
 
                 shellExecuteInfo.fMask |= NativeMethods.SEE_MASK_FLAG_DDEWAIT;
-
-                if (!NativeMethods.ShellExecuteEx(shellExecuteInfo))
-                {
-                    int errorCode = Marshal.GetLastWin32Error();
-                    if (errorCode == 0) 
-                    {
-                        switch ((long)shellExecuteInfo.hInstApp)
-                        {
-                            case NativeMethods.SE_ERR_FNF: errorCode = NativeMethods.ERROR_FILE_NOT_FOUND; break;
-                            case NativeMethods.SE_ERR_PNF: errorCode = NativeMethods.ERROR_PATH_NOT_FOUND; break;
-                            case NativeMethods.SE_ERR_ACCESSDENIED: errorCode = NativeMethods.ERROR_ACCESS_DENIED; break;
-                            case NativeMethods.SE_ERR_OOM: errorCode = NativeMethods.ERROR_NOT_ENOUGH_MEMORY; break;
-                            case NativeMethods.SE_ERR_DDEFAIL:
-                            case NativeMethods.SE_ERR_DDEBUSY:
-                            case NativeMethods.SE_ERR_DDETIMEOUT: errorCode = NativeMethods.ERROR_DDE_FAIL; break;
-                            case NativeMethods.SE_ERR_SHARE: errorCode = NativeMethods.ERROR_SHARING_VIOLATION; break;
-                            case NativeMethods.SE_ERR_NOASSOC: errorCode = NativeMethods.ERROR_NO_ASSOCIATION; break;
-                            case NativeMethods.SE_ERR_DLLNOTFOUND: errorCode = NativeMethods.ERROR_DLL_NOT_FOUND; break;
-                            default: errorCode = (int)shellExecuteInfo.hInstApp; break;
-                        }
-                    }
-
-                    if(errorCode == NativeMethods.ERROR_BAD_EXE_FORMAT || errorCode == NativeMethods.ERROR_EXE_MACHINE_TYPE_MISMATCH)
-                    {
-                        throw new Win32Exception(errorCode, "InvalidApplication");
-                    }
-                    
-                    throw new Win32Exception(errorCode);
-                }
+                Thread thread = new Thread(InvokeShellExecuteEx);
+                thread.SetApartmentState(System.Threading.ApartmentState.STA);
+                thread.Start(shellExecuteInfo);
+                thread.Join();
```

Comment:
The Win32Exception thrown from the new thread will probably be ignored. We should handle the error in the main thread.
Please take a look at the .NET implementation at https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Process.cs,2239
The `ShellExecuteOnSTAThread` method is defined at https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Process.cs,2856